contentcuration/contentcuration/management/commands/exportchannel.py

```diff
@@ -301,23 +315,28 @@ def create_associated_file_objects(kolibrinode, ccnode):
 def create_perseus_exercise(ccnode, kolibrinode, exercise_data, user_id=None):
     logging.debug("Creating Perseus Exercise for Node {}".format(ccnode.title))
     filename = "{0}.{ext}".format(ccnode.title, ext=file_formats.PERSEUS)
-    with tempfile.NamedTemporaryFile(suffix="zip", delete=False) as tempf:
-        create_perseus_zip(ccnode, exercise_data, tempf)
-        file_size = tempf.tell()
-        tempf.flush()
-
-        ccnode.files.filter(preset_id=format_presets.EXERCISE).delete()
-
-        assessment_file_obj = ccmodels.File.objects.create(
-            file_on_disk=File(open(tempf.name, 'r'), name=filename),
-            contentnode=ccnode,
-            file_format_id=file_formats.PERSEUS,
-            preset_id=format_presets.EXERCISE,
-            original_filename=filename,
-            file_size=file_size,
-            uploaded_by_id=user_id,
-        )
-        logging.debug("Created exercise for {0} with checksum {1}".format(ccnode.title, assessment_file_obj.checksum))
+    temppath = None
+    try:
+        with tempfile.NamedTemporaryFile(suffix="zip", delete=False) as tempf:
+            temppath = tempf.name
+            create_perseus_zip(ccnode, exercise_data, tempf)
+            file_size = tempf.tell()
+            tempf.flush()
+
+            ccnode.files.filter(preset_id=format_presets.EXERCISE).delete()
+
+            assessment_file_obj = ccmodels.File.objects.create(
+                file_on_disk=File(open(temppath, 'r'), name=filename),
+                contentnode=ccnode,
+                file_format_id=file_formats.PERSEUS,
+                preset_id=format_presets.EXERCISE,
+                original_filename=filename,
+                file_size=file_size,
+                uploaded_by_id=user_id,
+            )
+            logging.debug("Created exercise for {0} with checksum {1}".format(ccnode.title, assessment_file_obj.checksum))
+    finally:
+        temppath and os.unlink(temppath)
```

Comment:
woah! unlink!  sounds pretty serious

Reply:
Yeah, it's something to do with tempfiles needing to be closed to subsequently read from them. However, closing them deletes them, so I'm doing it manually here